UnityProject/Assets/Scripts/Managers/MatrixManager/MatrixManager.cs

```diff
@@ -34,18 +34,7 @@ public partial class MatrixManager : MonoBehaviour
 {
 	private static MatrixManager matrixManager;
 
-	public static MatrixManager Instance
-	{
-		get
-		{
-			if (matrixManager == null)
-			{
-				matrixManager = FindObjectOfType<MatrixManager>();
-			}
-
-			return matrixManager;
-		}
-	}
+	public static MatrixManager Instance => matrixManager;
```

Comment:
I think FindObjectOfType is required for the singleton monobehaviours

Reply:
nope

Reply:
Most managers should be moved over to this system, as checking FindObjectOfType. But the main reason for me switching it up was the null check on the get is very bad.

Reply:
Let's wait on what @Bod9001 has to say before marking as resolved.

Reply:
This should work okay as long as it's not used before Awake, you can use the initialisation manager to ensure that it is so